test/src/e2e/informer.test.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import { expect } from "chai";
+import "mocha";
+import { wait } from "../helper/promise";
+import CodeChain from "../helper/spawn";
+
+describe("Informer", function() {
+    let nodeA: CodeChain;
+    let nodeB: CodeChain;
+    const WebSocket = require("ws");
+    const address = "127.0.0.1";
+    before(async function() {
+        nodeA = new CodeChain();
+        nodeB = new CodeChain();
+        await Promise.all([nodeA.start(), nodeB.start()]);
+    });
+    describe("Hot Subscription", function() {
+        it("Client Subscribed", async function() {
+            const message1 = {
+                jsonrpc: "2.0",
+                result: " ",
+                id: "1"
+            };
+            const subscription = new WebSocket(
+                `ws://localhost:${nodeA.informerPort}`
+            );
+            subscription.addEventListener("open", () => {
+                subscription.send(
+                    JSON.stringify({
+                        jsonrpc: "2.0",
+                        id: 1,
+                        method: "register",
+                        params: []
+                    })
+                );
+            });
+            subscription.once("message", (message: any) => {
+                expect(message).include(JSON.stringify(message1));
```

Comment:
If you use expect in a callback, you cannot check the case when the callback is not called.
For the case, we are using promiseExpect.

How to use it:
1. Declare promiseExpect variable at the start of describe. [see](https://github.com/CodeChain-io/foundry/blob/master/test/src/e2e.long/futureTransaction.test.ts#L23)
2. Wrap promise using promiseExpect.shouldFulfill(). [see](https://github.com/CodeChain-io/foundry/blob/master/test/src/e2e.long/futureTransaction.test.ts#L58)
3. Call promiseExpect.checkFulfilled(); [see](https://github.com/CodeChain-io/foundry/blob/master/test/src/e2e.long/futureTransaction.test.ts#L65)

The checkFulfilled() will throw an error if there are some promises that are not fulfilled.

Reply:
I recommend you read PromiseExpect code. It is simple.